### PR TITLE
fix eslint

### PR DESCRIPTION
**Pull request by Rebal67**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
-        "no-undef": "off",
-        "no-unused-vars": "off",
         "indent": ["error", 4],
         "camelcase": "error",
         "prettier/prettier": "error",
-        "@typescript-eslint/explicit-module-boundary-types": "off",
-        "@typescript-eslint/explicit-function-return-type": "off",
-        "@typescript-eslint/no-explicit-any": "error",
-        "@typescript-eslint/no-unused-vars": "warn"
+        "react/display-name": "off",
```
Comment (theblockstalk) [inline .eslintrc.json:11]: why react rules in the SDK?

File: package.json [hunk 2]
```diff
@@ -25,12 +25,6 @@
       "pre-commit": "tsdx lint"
     }
   },
-  "prettier": {
```
Comment (theblockstalk) [inline package.json:28]: ahhh, did not know these were in package.json. I thought they were only in `.prettierrc.json`. Perhaps we should check the other repos for this. Maybe this is why there were/are conflicts...

File: .eslintrc.json (hunk 1 above)
Comment (Rebal67) [reply .eslintrc.json:11]: i copied the other files settings to make sure everything same forgot to remove this line

File: .eslintrc.json [hunk 3]
```diff
@@ -1,31 +1,25 @@
...
+    "plugins": ["@typescript-eslint","prettier"],
     "rules": {
         "eqeqeq": "error",
         "no-console": "warn",
-        "no-undef": "off",
-        "no-unused-vars": "off",
         "indent": ["error", 4],
         "camelcase": "error",
         "prettier/prettier": "error",
-        "@typescript-eslint/explicit-module-boundary-types": "off",
```
[the remark this answers is not in the document]
Comment (Rebal67) [reply .eslintrc.json:20]: deleted by mistake

File: package.json (hunk 2 above)
Comment (Rebal67) [reply package.json:28]: all checked there is an easlint rules in id-demo that should be removed

File: .eslintrc.json (hunk 1 above)
Comment (theblockstalk) [reply .eslintrc.json]: I think you need to remove all of these react rules

File: .eslintrc.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (theblockstalk) [reply .eslintrc.json]: can you please re-add

File: package.json (hunk 2 above)
Comment (theblockstalk) [reply package.json:28]: please make a PR for that as well

File: .eslintrc.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Rebal67) [reply .eslintrc.json]: added

File: .eslintrc.json (hunk 1 above)
Comment (Rebal67) [reply .eslintrc.json]: removed

**Pull request merged**